mocked-api/handlers.ts

```diff
@@ -435,6 +435,124 @@ export const handlers = [
       )
     );
   }),
+  // update a processor
+  rest.put(
+    `${apiUrl}/bridges/:bridgeId/processors/:processorId`,
+    (req, res, ctx) => {
+      const { bridgeId, processorId } = req.params;
+      const { name, filters, transformationTemplate, source, action } =
+        req.body as ProcessorRequest;
+
+      const existingBridge = db.bridge.findFirst({
+        where: {
+          id: {
+            equals: bridgeId as string,
+          },
+        },
+      });
+
+      if (!existingBridge) {
+        return res(
+          ctx.status(404),
+          ctx.delay(apiDelay),
+          ctx.json({
+            ...error_not_found,
+            reason: `Bridge with id '${
+              bridgeId as string
+            }' for customer 'XXXXXXXX' does not exist`,
+          })
+        );
+      }
+
+      const processor = db.processor.findFirst({
+        where: {
+          id: {
+            equals: processorId as string,
+          },
+        },
+      });
+
+      if (!processor) {
+        return res(
+          ctx.status(404),
+          ctx.delay(apiDelay),
+          ctx.json({
+            ...error_not_found,
+            reason: `Processor with id '${
+              processorId as string
+            }' for customer 'XXXXXXXX' does not exist`,
+          })
+        );
+      }
+
+      const existingProcessors = db.processor.findMany({
+        where: {
+          name: {
+            equals: name,
+          },
+          bridge: {
+            id: {
+              equals: bridgeId as string,
+            },
+          },
+        },
+      });
+
+      const processorNameCollision =
+        existingProcessors.length === 1 &&
+        existingProcessors[0].id !== processorId;
+      if (processorNameCollision || existingProcessors.length > 1) {
```

Comment:
Out of curiosity, is it possible that `existingProcessors  > 1`? If the query is on the name, which is unique, `existingProcessors` length can only be 1 or 0.

Reply:
It is possible only if existing data are malformed. So, yes, it eventually could.